dashboard/app/controllers/admin_reports_controller.rb

```diff
@@ -68,12 +67,12 @@ def funometer_by_script_level
     @level_id = params[:level_id]
     @level_name = Level.where('id = ?', @level_id).pluck(:name)[0]
 
-    # Generate the funometer percentages for the level, by day.
     ratings = PuzzleRating.where('script_id = ?', @script_id).where('level_id = ?', @level_id)
-    ratings_by_day = ratings.group('DATE(created_at)').order('DATE(created_at)')
-    percentages_by_day = ratings_by_day.pluck('DATE(created_at)', '100.0 * SUM(rating) / COUNT(rating) AS percentage')
+    @overall_percentage = get_percentage_positive(ratings)
+
+    # Generate the funometer percentages for the level, by day, for the last month.
     @ratings_by_day_headers = ['Date', 'Percentage', 'Count']
-    @ratings_by_day = ratings_by_day.select('DATE(created_at) AS day', '100.0 * SUM(rating) / COUNT(rating) AS percentage', 'COUNT(rating) AS cnt')
+    @ratings_by_day, percentages_by_day = get_ratings_by_day(ratings)
 
     render locals: {percentages_by_day: percentages_by_day.to_a.map{|k,v|[k.to_s,v.to_f]}}
```

Comment:
There isn't good rhyme or reason, I'll clean that up in another PR, I want to get this one in for today's DTP.